src/node_contextify.cc

```diff
@@ -1099,12 +1099,23 @@ void ContextifyContext::CompileFunction(
           env->cached_data_string(),
           buf.ToLocalChecked()).IsNothing()) return;
     }
+
+    // Report if cache was produced.
     if (fun->Set(
         parsing_context,
         env->cached_data_produced_string(),
         Boolean::New(isolate, cached_data_produced)).IsNothing()) return;
   }
 
+  // Report if cache was rejected.
+  if (options == ScriptCompiler::kConsumeCodeCache) {
+    if (fun->Set(parsing_context,
+                 env->cached_data_rejected_string(),
+                 Boolean::New(isolate, source.GetCachedData()->rejected))
+            .IsNothing())
+      return;
+  }
```

Comment:
@ryzokuken 👆 That's a neat way to handle this. What do you think about having `compileFunction` return a results object with a `value` property (kind of like the generator protocol) but with other properties like the `cachedDataRejected` hanging off it (so it doesn't have to hang off the generated function).

Reply:
@jdalton that's actually what I had originally intended to do when I was designing `vm.compileFunction`, but chose to do things this way for the sake of consistency (all related functions in the vm module seem to do things this way?).

I'm still not opposed to the idea of doing things that way, if enough people agree, but let's just keep in mind that that'll have to happen as a semver-major change.

Do you want me to make an issue to track people's opinions on the idea and follow it up with a PR if everyone agrees?

Reply:
@ryzokuken 
> Do you want me to make an issue to track people's opinions on the idea and follow it up with a PR if everyone agrees?

That would be great, thank you!